model/Asset.ts

```diff
@@ -0,0 +1,43 @@
+type AssetInit = {
+  id: string
+  name: string
+  description?: string
+  domain?: string
+  updateFrequency?: string
+  tags?: DataMarketplace.Tag[]
+  createdAt?: Date | number | string
+  updatedAt?: Date | number | string
+}
+
+export class Asset implements DataMarketplace.Asset {
```

Comment:
More object orientation. I think we need to have a talk about the programming paradigm.